launch_testing_ament_cmake/cmake/add_launch_test.cmake

```diff
@@ -122,6 +122,20 @@ function(add_launch_test filename)
     "--package-name=${PROJECT_NAME}"
   )
 
+  if(NOT DEFINED ENV{ROS_DOMAIN_ID})
+    # In the absence of a ROS_DOMAIN_ID, launch_testing will pick a ROS_DOMAIN_ID
+    # that is not in use by another launch_testing instance running on this machine.
+    # This allows tests to run in parallel.  If a ROS_DOMAIN_ID is already set,
+    # launch_testing will use that ROS_DOMAIN_ID.  In this case, tests must be designed
+    # carefully to not interfere with one-another, or must not be run in parallel
+    # The --isolated flag cannot provide isolation between multiple hosts
+    # Note that the ROS_DOMAIN_ID is checked when `colcon build` is run
+    set(cmd
+      "${cmd}"
+      "--isolated"
+    )
+  endif()
+
```

Comment:
Since this is determining the choice at CMake configure time it would be wrong if the env var is not set when building the package but set later when actually running the test. Therefore I think this check should be moved into the Python script.

Reply:
Does the --isolated argument still make sense then?  Is there a use case to force isolation to be turned on when ROS domain ID is set, or would the --isolation flag only do something when no domain ID is already set?

I guess we could also change the flag to be --isolated=0 or --isolated=1 to force it on or off, and you get the automatic behavior if you don't pass the flag.

Reply:
I would be fine to not have any command line option and make the behavior only depend on the presence of the environment variable.

If you would like to have a command line option to explicitly control the behavior that is fine too. Maybe two mutually exclusive options would be most readable (since `isolated` doesn't convey much about the *how*):

* `--domain-id` (`type=int`) (since the very same could be achieve by setting the env var I am not sure how valuable this one is...)
* `--random-domain-id` (`action=store_true`)